### ISSUE TITLE
Progression patching affecting progression features added into base class

### ISSUE THREAD

**Issue by JohN100x1** [JohN100x1/WOTR_IsekaiMod#117]
[User reported bug](https://forums.nexusmods.com/index.php?/topic/12204503-isekai-mod/?p=122591994):

> I'm having trouble with this mod (version 4.2.0) + TableTopTweaks (Base+Core+Tweaks) when it comes to selecting the Capstone feat 'Perfect Body, Flawless Mind'. Normally when selecting this feat you are shown eight more pages on the character level up screen to select which ability scores you wish to increase. With this mod installed the additional pages are not activated so you can't select the ability scores to increase. This happens on any class I try to level up as well as the legacies used by the Isekai classes.
> 
> Has anyone else got this problem?

I have tested this in two cases:
- Selecting the `Perfect Body, Flawless Mind` as a level 20 barbarian causes the bug to occur.
- Selecting the `Perfect Body, Flawless Mind` as a level 20 isekai protagonist works as normal.

I believe this was caused by the [progression patching](https://github.com/JohN100x1/WOTR_IsekaiMod/blob/49ec3901b1d10cbe9ed551bd8151ea14d009a2e8/IsekaiMod/Utilities/StaticReferences.cs#L218).
Because `Perfect Body, Flawless Mind` is a progression, it was patched so that the Isekai protagonist class was added to `m_Classes`.
This means when the user selects this progression feature as a level 20 barbarian, they will not get any features from the `Perfect Body, Flawless Mind` progression because their Isekai Protagonist class level is zero.

I confirmed this was the exact cause by executing this line after patching, which causes it to work as normal.
```
var perfectbody = BlueprintTools.GetBlueprint<BlueprintProgression>("cb61beef4f354992a09e821d170a6582");
perfectbody.m_Classes = new BlueprintProgression.ClassWithLevel[0];
```

**Comment by kjk001** [comment]
Argh I think I know what causes it.
Perfect Body, Flawless Mind wasn't class locked at all was it?
That would mean that if we add just a single class it would lock against all others.

Ok fairly sure I know where the check if a progression was class locked before patching it is missing.

**Comment by kjk001** [comment]
@JohN100x1 
I added a check if the progression was class locked on my branch, I still need to test it though even if I am 99% certain it will work.
I will do that later.
Also fairly sure only progressions added by other mods were affected because the basegame always class locks them even when it is not necessary or actually seems counterproductive...

**Comment by kjk001** [comment]
@JohN100x1 tested barbarian and it now works again, not sure if you want to wait for patch 4.3 to release it or if I should do a rebase to 4.2 release and apply it and the pala fix there for a version 4.2.1

**Comment by JohN100x1** [comment]
Let's leave it for 4.3. I think we can get the release done by week.
